swift-sdk/Internal/IterableHtmlMessageViewController.swift

```diff
@@ -154,30 +174,95 @@ class IterableHtmlMessageViewController: UIViewController {
         return webView as WebViewProtocol
     }
     
+    private static func trackClickOnDismiss(internalAPI: IterableAPIInternal?,
+                                            params: Parameters,
+                                            futureClickedURL: Promise<URL, IterableError>,
+                                            withURL url: URL,
+                                            andDestinationURL destinationURL: String) {
+        ITBInfo()
+        futureClickedURL.resolve(with: url)
+        if let messageMetadata = params.messageMetadata {
+            internalAPI?.trackInAppClick(messageMetadata.message,
+                                         location: messageMetadata.location,
+                                         inboxSessionId: params.inboxSessionId,
+                                         clickedUrl: destinationURL)
+        }
+    }
+
     /// Resizes the webview based upon the insetPadding, height etc
     private func resizeWebView() {
         let parentPosition = ViewPosition(width: view.bounds.width,
                                           height: view.bounds.height,
                                           center: view.center)
         IterableHtmlMessageViewController.calculateWebViewPosition(webView: webView,
-                                                                   safeAreaInsets: IterableHtmlMessageViewController.safeAreaInsets(for: view),
+                                                                   safeAreaInsets: InAppCalculations.safeAreaInsets(for: view),
                                                                    parentPosition: parentPosition,
                                                                    paddingLeft: parameters.padding.left,
                                                                    paddingRight: parameters.padding.right,
                                                                    location: location)
             .onSuccess { [weak self] position in
-                self?.webView.set(position: position)
+                if self?.parameters.isModal == true && self?.parameters.shouldAnimate == true {
+                    self?.animateWhileEntering(position: position)
+                } else {
+                    self?.webView.set(position: position)
+                }
             }
     }
     
-    private static func safeAreaInsets(for view: UIView) -> UIEdgeInsets {
-        if #available(iOS 11, *) {
-            return view.safeAreaInsets
-        } else {
-            return .zero
+    private func animateWhileEntering(position: ViewPosition) {
+        Self.animate(duration: parameters.animationDuration) { [weak self] in
```

Comment:
Huh. I've never seen `Self` before. Is it just the same thing as `self`? Seems like it.

Reply:
`Self` refers to class methods and vars, whereas `self` is for instance.

Reply:
OH. That makes way more sense!